### PR TITLE
User persistence

### LINKED ISSUE #29: Create Firebase databases and store/load dummy data from the app to be used for the stats display.
Create two databases: one for Games and one for Users.

image

- Games: Store the results of played games and their associated user IDs.
- Users: Store lists of all game IDs played by each user and additional statistical data.

### PR DESCRIPTION
**Pull request by MaelArrive**
When you sign up, a user is created in the Realtime Database in Firebase, with all attribute values set to 0 (currentStreak, maxStreak, averageRank, averageGuesses, averageTime, timesPlayed).

Upon logging in, the application fetches the user information from the database and updates the UserModel accordingly.

You can view the statistics by clicking on the "Test the statistics" button. From this page, you can populate the user in the database by clicking on the "Populate stats" button (this is simply to set some dummy values other than 0 to visualize that the feature is functioning).

When you log out, the UserModel is reset.

Closes #29.

### CODE REVIEW

File: utilities/Utils.ts [hunk 1]
```diff
@@ -28,6 +30,16 @@ export class Utils {
...
+    /**
+     * Get random user model
+     * @returns a random user model
+     */
+    public static getRandomUserModel(): UserModel {
+        const user = new UserModel();
+        user.updateStats(321, 2321, 345, 435, 324, 2432)
+        return user;
+    }
+
```
Comment (roxannecvl) [inline utilities/Utils.ts:42]: Is this function for testing purposes only ? You should add a TODO : remove after testing if so

**Review by roxannecvl — CHANGES_REQUESTED** [review]
- Can't click on populate stats 
- When running npm install wa now get : 1 moderate severity vulnerability
please look into that
- I'd like to have the GitHub button on the right like before and on its left the login button. 
- The button should say Log In (if not logged in, people without an account will 
understand that t they should also click there you don't need sign up/ log in in my opinion) 
- If possible make the button say Log out when a user is logged in, shouldn't be too hard :)

**Comment by MaelArrive** [comment]
> * When running npm install we now get : 1 moderate severity vulnerability
>   please look into that

Just run ```npm audit fix``` to address all issues

**Review by roxannecvl — APPROVED** [review]
All good, don't forget to add a presenter when cleaning the architecture of our project :)

**Review by martinctl — APPROVED** [review]

**Pull request merged**